align_system/algorithms/oracle_adm.py

```diff
@@ -50,42 +51,58 @@ def _inverse_distance(self, target_kdma_values, system_kdmas, min_value=0., max_
         # Small epsilon for a perfect (0 distance) match
         return math.sqrt(1/(distance+1e-16))
 
-    def choose_action(self, scenario_state, available_actions, alignment_target, **kwargs):
+    def choose_action(self, scenario_state, available_actions, alignment_target, 
+                      distribution_matching: str='sample', kde_norm='globalnorm', 
+                      **kwargs):
         if available_actions is None or len(available_actions) == 0:
             return None
 
         if alignment_target is None:
             raise ValueError("Oracle ADM needs alignment target")
 
-        # Build out the corresponding kdma_association target dictionary
-        target_kdma_assoc = {
-            target['kdma']: target['value']
-            for target in alignment_target.kdma_values
-        }
-
-        # Weight action choices with distance-based metric
-        inv_dists = [
-            self._inverse_distance(target_kdma_values=target_kdma_assoc, system_kdmas=action.kdma_association)
-            for action in available_actions
-        ]
-
-        # Convert inverse distances to probabilities
-        probs = [inv_dist/sum(inv_dists) for inv_dist in inv_dists]
-
-        if self.probabilistic:
-            action_to_take = np.random.choice(available_actions, p=probs)
-        else:  # Always choose (one of the) max probability action
-            max_prob = max(probs)
-            max_actions = [idx for idx, p in enumerate(probs) if p == max_prob]
-            action_to_take = available_actions[random.choice(max_actions)]
-
-        # Log scoring results
-        results = pd.DataFrame([
-            (action.unstructured, prob)
-            for action, prob in zip(available_actions, probs)
-        ], columns=["choice", "probability"])
-        results = results.sort_values(by=["probability"], ascending=False)
-        log.explain(results)
+        # Check if targets are scalar
+        scalar_target_kdma_assoc = {}
+        for target in alignment_target.kdma_values:
+            if 'value' in target and target['value'] is not None:
+                scalar_target_kdma_assoc[target['kdma']] = target['value']
```

Comment:
Same nit comment as previous PR: This doesn't account for if there is a mix of scalar and distribution targets, that's not something we need to address now -- but a comment warning about that would be nice